cmd/hubagent/workload/setup.go

```diff
@@ -198,7 +198,7 @@ func SetupControllers(ctx context.Context, wg *sync.WaitGroup, mgr ctrl.Manager,
 		klog.Info("Setting up clusterResourcePlacement watcher")
 		if err := (&clusterresourceplacementwatcher.Reconciler{
 			PlacementController: clusterResourcePlacementControllerV1Beta1,
-		}).SetupWithManager(mgr); err != nil {
+		}).SetupWithManagerForClusterResourcePlacement(mgr); err != nil {
```

Comment:
this seems to indicate that we will have two reconcilers? One for CRP and one for RP?

Reply:
they're independent.

There are two options:
1. one CRP watcher, one RP watcher and both enqueue the key to the (C)RP reconciler.
2. CRP and RP have separate reconciler and watcher enqueues the keys separately.

For now, these two options have not much diff, unless they want to share the same view/memory/etc and they have to be a single reconciler.